util/src/main/java/io/zeebe/util/health/FailureListener.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.0. You may not use this file
+ * except in compliance with the Zeebe Community License 1.0.
+ */
+package io.zeebe.util.health;
+
+public interface FailureListener {
+  void onFailure();
```

Comment:
Could it make sense to pass the failure itself (e.g. `Throwable` I guess?), and the component which triggered this? That way implementations can be react on different failures, and also be reused for different components.

Reply:
Since we use this only to report failures, I think this is enough for now. It would be useful if we are reacting to failures. But now I am not sure what information is needed or how we will react to them. I believe the interface will change anyway in future.